ArduCopter/mode_rc_car.cpp

```diff
@@ -0,0 +1,1000 @@
+            target_climb_rate = get_avoidance_adjusted_climbrate(target_climb_rate);
+
+            // set position controller targets adjusted for pilot input
+            takeoff.do_pilot_takeoff(target_climb_rate);
+            break;
+
+        case AltHold_Landed_Ground_Idle:
+            attitude_control->reset_yaw_target_and_rate();
+            FALLTHROUGH;
+
+        case AltHold_Landed_Pre_Takeoff:
+            attitude_control->reset_rate_controller_I_terms_smoothly();
+            pos_control->relax_z_controller(0.0f);   // forces throttle output to decay to zero
+            break;
+
+        case AltHold_Flying:
+            copter.motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
+
+            // get avoidance adjusted climb rate
+            target_climb_rate = get_avoidance_adjusted_climbrate(target_climb_rate);
+
+            // update the vertical offset based on the surface measurement
+            copter.surface_tracking.update_surface_offset();
+
+            // Send the commanded climb rate to the position controller
+            // pos_control->set_pos_target_z_from_climb_rate_cm(target_climb_rate);
+            if (copter.g.rccar_unlimited_height == 0) {
+                float alt_target = channel_throttle->get_control_in() * 0.2;	
+                pos_control->set_pos_target_z_cm(alt_target);	
+            } else {	
+                pos_control->set_pos_target_z_from_climb_rate_cm(target_climb_rate);	
+            }
+            break;
+        }
+
+        // flowhold attitude target calculations
+        Vector2f bf_angles;
+
+        // calculate alt-hold angles
+        int16_t roll_in = copter.channel_roll->get_control_in();
+        int16_t pitch_in = copter.channel_pitch->get_control_in();
+        float angle_max = copter.aparm.angle_max;
+        get_pilot_desired_lean_angles(bf_angles.x, bf_angles.y, angle_max, attitude_control->get_althold_lean_angle_max_cd());
+
+        if (quality_filtered >= flow_min_quality &&
+            AP_HAL::millis() - copter.arm_time_ms > 3000) {
+            // don't use for first 3s when we are just taking off
+            Vector2f flow_angles;
+
+            flowhold_flow_to_angle(flow_angles, (roll_in != 0) || (pitch_in != 0));
+            flow_angles.x = constrain_float(flow_angles.x, -angle_max/2, angle_max/2);
+            flow_angles.y = constrain_float(flow_angles.y, -angle_max/2, angle_max/2);
+            bf_angles += flow_angles;
+        }
+        bf_angles.x = constrain_float(bf_angles.x, -angle_max, angle_max);
+        bf_angles.y = constrain_float(bf_angles.y, -angle_max, angle_max);
+
+    #if AC_AVOID_ENABLED == ENABLED
+        // apply avoidance
+        copter.avoid.adjust_roll_pitch(bf_angles.x, bf_angles.y, copter.aparm.angle_max);
+    #endif
+
+        // call attitude controller
+        copter.attitude_control->input_euler_angle_roll_pitch_euler_rate_yaw(bf_angles.x, bf_angles.y, target_yaw_rate);
```

Comment:
Internal error may be coming from one of the calls to input_euler_angle_roll_pitch_euler_rate_yaw